models/contributors/contributors_graph.go

```diff
@@ -0,0 +1,114 @@
+// Copyright 2023 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package contributors
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	repo_model "code.gitea.io/gitea/models/repo"
+	user_model "code.gitea.io/gitea/models/user"
```

Comment:
models/* should not depend on modules/* ... it's the other way around :)

this package / func in this file ... is directly used by web and api ... so it should go into services/*

Reply:
(https://docs.gitea.com/contributing/guidelines-backend#packages-list)

Reply:
> models/* should not depend on modules/*

Erm… what? How sure are you about your claim? It sounds like BS to me. (https://docs.gitea.com/contributing/guidelines-backend#package-dependencies)

> it should go into services/*

That I agree with, the file defines no database query but only uses existing ones.

Reply:
Should be fixed with d6887e4